src/main/java/org/jabref/gui/preferences/customization/CustomizationTabViewModel.java

```diff
@@ -45,4 +108,58 @@ public BooleanProperty useCustomDOIProperty() {
     public StringProperty useCustomDOINameProperty() {
         return this.useCustomDOINameProperty;
     }
+
+    public ListProperty<CustomApiKeyPreferences> customApiKeyPrefsProperty() {
+        return this.customApiKeyPrefsListProperty;
+    }
+
+    public ObjectProperty<CustomApiKeyPreferences> selectedCustomApiKeyPrefProperty() {
+        return this.selectedCustomApiKeyPrefProperty;
+    }
+
+    public BooleanProperty useCustomApiKeyProperty() {
+        return this.useCustomApiKeyProperty;
+    }
+
+    public StringProperty customApiKeyText() {
+        return this.customApiKeyTextProperty;
+    }
+
+    public void checkCustomApiKey() {
+        final String apiKeyName = selectedCustomApiKeyPrefProperty.get().getName();
+        final String testUrlWithoutApiKey = API_KEY_NAME_URL.get(apiKeyName);
+        final String apiKey = customApiKeyTextProperty.get();
+
+        final String connectionSuccessText = Localization.lang("Connection successful!");
+        final String connectionFailedText = Localization.lang("Connection failed!");
+        final String dialogTitle = Localization.lang("Check %0 API Key Setting", apiKeyName);
```

Comment:
These strings seem not to be used more than once. You should inline them.

Reply:
fixed.